### ISSUE TITLE
"index out of bounds"`

### ISSUE THREAD

**Issue by Chen-PengF** [locuslab/trellisnet#8]
Weight normalization applied
/word_WT103/splitcross.py:130: UserWarning: Implicit dimension choice for log_softmax has been deprecated. Change the call to include dim=X as an argument.
  softmaxed_all_head_res = torch.nn.functional.log_softmax(all_head_res)
/TrellisNet/word_WT103/splitcross.py:67: UserWarning: Implicit dimension choice for log_softmax has been deprecated. Change the call to include dim=X as an argument.
  tail_entropy = torch.nn.functional.log_softmax(tail_res)
/TrellisNet/word_WT103/splitcross.py:164: UserWarning: Implicit dimension choice for log_softmax has been deprecated. Change the call to include dim=X as an argument.
  tail_entropy = torch.gather(torch.nn.functional.log_softmax(tail_res), dim=1, index=indices).squeeze()
/opt/conda/conda-bld/pytorch_1614378124864/work/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [2,0,0], thread: [32,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
/opt/conda/conda-bld/pytorch_1614378124864/work/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [2,0,0], thread: [34,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
/opt/conda/conda-bld/pytorch_1614378124864/work/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [2,0,0], thread: [35,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.

**Comment by Chen-PengF** [comment]
why is this?

**Comment by gitdog-0110** [comment]
> Weight normalization applied /word_WT103/splitcross.py:130: UserWarning: Implicit dimension choice for log_softmax has been deprecated. Change the call to include dim=X as an argument. softmaxed_all_head_res = torch.nn.functional.log_softmax(all_head_res) /TrellisNet/word_WT103/splitcross.py:67: UserWarning: Implicit dimension choice for log_softmax has been deprecated. Change the call to include dim=X as an argument. tail_entropy = torch.nn.functional.log_softmax(tail_res) /TrellisNet/word_WT103/splitcross.py:164: UserWarning: Implicit dimension choice for log_softmax has been deprecated. Change the call to include dim=X as an argument. tail_entropy = torch.gather(torch.nn.functional.log_softmax(tail_res), dim=1, index=indices).squeeze() /opt/conda/conda-bld/pytorch_1614378124864/work/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [2,0,0], thread: [32,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed. /opt/conda/conda-bld/pytorch_1614378124864/work/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [2,0,0], thread: [34,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed. /opt/conda/conda-bld/pytorch_1614378124864/work/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [2,0,0], thread: [35,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.

Hello, have you solved your problem? I also encountered the same problem, and the error also occurred in this function with torch.gather
